services/addons/tests/firmware_manager/test_firmware_manager.py

```diff
@@ -595,17 +779,180 @@ def test_check_for_upgrades(mock_upgrade_limit, mock_stfq, mock_logging):
             call("Adding '8.8.8.8' to the firmware manager upgrade queue"),
             call("Firmware upgrade successfully started for '8.8.8.8'"),
             call("Adding '9.9.9.9' to the firmware manager upgrade queue"),
-            call("Firmware upgrade successfully started for '9.9.9.9'")
+            call("Firmware upgrade successfully started for '9.9.9.9'"),
         ]
     )
-    mock_logging.info.assert_called_with(
-        "Firmware upgrade successfully started for '9.9.9.9'"
-    )
 
 
 # Other tests
 
 
+@patch("addons.images.firmware_manager.firmware_manager.pgquery.query_db")
+def test_was_latest_ping_successful_for_rsu(mock_query_db):
+    # prepare
+    rsu_ip = "8.8.8.8"
+    expected_query = "select result from ping where rsu_id=(select rsu_id from rsus where ipv4_address='8.8.8.8') order by timestamp desc limit 1"
+    mock_query_db.return_value = [(True,)]
+
+    # execute
+    result = firmware_manager.was_latest_ping_successful_for_rsu(rsu_ip)
+
+    # verify
+    assert result == True
+    mock_query_db.assert_called_with(expected_query)
+
+
+@patch(
+    "addons.images.firmware_manager.firmware_manager.does_consecutive_failure_count_exist_for_rsu"
+)
+@patch("addons.images.firmware_manager.firmware_manager.pgquery.write_db")
+def test_increment_consecutive_failure_count_for_rsu_record_exists(
+    mock_write_db, mock_does_consecutive_failure_count_exist_for_rsu
+):
+    # prepare
+    mock_does_consecutive_failure_count_exist_for_rsu.return_value = True
+    rsu_ip = "8.8.8.8"
+    expected_query = "update consecutive_firmware_upgrade_failures set consecutive_failures=consecutive_failures+1 where rsu_id=(select rsu_id from rsus where ipv4_address='8.8.8.8')"

```

Comment:
question(non-blocking): This a general question about the tests in this file. I'm unfamiliar with testing for which query was executed. I'm more used to testing whether the correct data was returned by a function. What does testing for the specific query we run get us?

Reply:
Unit tests should depend solely on the class being tested. If we wanted to test with real data, we would need a live database running in the background. This test checks that the correct queries are executed with the expected parameters when the method is called. By validating this behavior without a live database, the test ensures the class logic works correctly in isolation.